app/services/defer_offer.rb

```diff
@@ -2,27 +2,36 @@ class DeferOffer
   include ActiveModel::Model
   include ImpersonationAuditHelper
 
+  attr_reader :actor, :application_choice, :course_option
+
   def initialize(actor:, application_choice:)
-    @auth = ProviderAuthorisation.new(actor: actor)
+    @actor = actor
     @application_choice = application_choice
+    @course_option = application_choice.current_course_option
   end
 
   def save!
-    @auth.assert_can_make_decisions!(application_choice: @application_choice, course_option_id: @application_choice.current_course_option.id)
+    auth.assert_can_make_decisions!(application_choice: application_choice, course_option: course_option)
 
-    audit(@auth.actor) do
-      prior_status = @application_choice.status
+    audit(actor) do
+      previous_status = application_choice.status
 
       ActiveRecord::Base.transaction do
-        ApplicationStateChange.new(@application_choice).defer_offer!
-        @application_choice.update!(
-          status_before_deferral: prior_status,
+        ApplicationStateChange.new(application_choice).defer_offer!
+        application_choice.update!(
+          status_before_deferral: previous_status,
           offer_deferred_at: Time.zone.now,
         )
       end
 
-      CandidateMailer.deferred_offer(@application_choice).deliver_later
-      StateChangeNotifier.call(:defer_offer, application_choice: @application_choice)
+      CandidateMailer.deferred_offer(application_choice).deliver_later
+      StateChangeNotifier.call(:defer_offer, application_choice: application_choice)
     end
   end
+
+private
+
+  def auth
+    @auth ||= ProviderAuthorisation.new(actor: actor)
```

Comment:
I vaguely remember that there's a side effect in memoizing `ProviderAuthorization` instances.
ie. Multiple (differing) auth checks won't update the internal `errors` after the first call. 
These services only seem to make one call to the `ProviderAuthorization` instance, so do we need to memoize and risk potential future confusion?

Reply:
I think the issue you are referring to was hapenning in one of the controllers where more than one permission checks where taking place but becaus of memoizing only the first call was execute - however I don't recall where that was hapenning.

However, happy to remove for clarity.

Reply:
I think you're right the bug was with multiple Auth checks, so perhaps not a concern here.